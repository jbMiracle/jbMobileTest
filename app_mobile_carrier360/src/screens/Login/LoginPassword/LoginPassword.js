import React, { Component } from 'react';
import { View, Alert } from 'react-native';
import { connectStyle, Text, Button } from 'native-base';
import PropTypes from 'prop-types';
import styleDef from '../LoginPassword/LoginPassword.style';
import C360LogoContainer from '../../../components/C360LogoContainer/C360LogoContainer';
import { C360InputField } from '../../../components/C360InputField/C360InputField';
import strings from '../../../services/localization';

const loginPasswordTextOptions = { scope: 'LoginPassword' };

class LoginPassword extends Component {
  state = {
    username: null,
    password: null,
  };
  forgotPasswordClicked = () => {
    this.props.navigation.navigate('ForgotPassword');
  };
  createAccountClicked = () => {
    Alert.alert(null, 'Create Account Clicked!');
  };
  loginClicked = () => {
    this.props.navigation.navigate('Main');
  };
  render() {
    return (
      <C360LogoContainer>
        <C360InputField
          placeHolderText={ strings('username', loginPasswordTextOptions) }
          leftIcon='jbh-Profile_Icon'
          dark
          testID='login_Username'
          onChangeText={ (text) => {
            this.setState({ username: text });
          } }
        />
        <C360InputField
          placeHolderText={ strings('password', loginPasswordTextOptions) }
          leftIcon='jbh-Password_2'
          dark
          secureTextEntry
          testID='login_Password'
          onChangeText={ (text) => {
            this.setState({ password: text });
          } }
        />
        <View style={ styleDef.forgotPasswordContainer }>
          <Text testID='forgotPassword' onPress={ this.forgotPasswordClicked }>
            <Text style={ styleDef.lightText }>{ strings('forgotPassword_1', loginPasswordTextOptions) }</Text>
            <Text style={ styleDef.semiBoldText }>{ strings('forgotPassword_2', loginPasswordTextOptions) }</Text>
            <Text style={ styleDef.lightText }>?</Text>
          </Text>
        </View>
        <Button
          style={ { flex: 1, marginVertical: 30 } }
          block
          primaryDark
          dark
          disabled={ !(this.state.username && this.state.password) }
          onPress={ this.loginClicked }
        >
          <Text>
            { strings('login_button', loginPasswordTextOptions) }
          </Text>
        </Button>
        <View style={ styleDef.createAccountContainer }>
          <Text testID='createAccount' onPress={ this.createAccountClicked }>
            <Text style={ styleDef.lightText }>{ strings('create_account_1', loginPasswordTextOptions) }</Text>
            <Text style={ styleDef.semiBoldText }>{ strings('create_account_2', loginPasswordTextOptions) }</Text>
          </Text>
        </View>
      </C360LogoContainer>
    );
  }
}
LoginPassword.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
    state: PropTypes.object.isRequired,
  }).isRequired,
};
export default connectStyle('C360.LoginPassword', styleDef)(LoginPassword);
