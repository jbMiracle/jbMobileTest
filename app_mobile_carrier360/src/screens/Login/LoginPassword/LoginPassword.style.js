import variables from '../../../themes/defaultTheme/variables/platform';
import C360Colors from '../../../themes/360Colors';

export default {
  lightText: {
    fontSize: variables.subTitleFontSize,
    fontFamily: variables.subtitleFontFamily,
    color: C360Colors.White,
  },
  semiBoldText: {
    fontSize: variables.subTitleFontSize,
    fontFamily: variables.titleFontfamily,
    color: C360Colors.White,
  },
  forgotPasswordContainer: {
    paddingLeft: 30,
    paddingTop: 10,
    alignSelf: 'flex-start',
  },
  createAccountContainer: {
    alignSelf: 'center',
  },
};
