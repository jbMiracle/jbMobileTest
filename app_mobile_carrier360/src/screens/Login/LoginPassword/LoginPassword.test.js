import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import LoginPassword from './LoginPassword';

describe('Render Login with Password screen', () => {
  it('Render Login with Password screen', () => {
    const renderer = new ShallowRenderer();
    const props = {
      navigation: {
        navigate: () => {},
        state: {},
      },
    };
    renderer.render(<LoginPassword { ...props } />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
