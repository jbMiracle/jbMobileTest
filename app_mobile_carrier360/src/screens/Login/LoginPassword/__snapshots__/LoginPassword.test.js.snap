// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Login with Password screen Render Login with Password screen 1`] = `
<LoginPassword
  navigation={
    Object {
      "navigate": [Function],
      "state": Object {},
    }
  }
  style={
    Object {
      "createAccountContainer": Object {
        "alignSelf": "center",
      },
      "forgotPasswordContainer": Object {
        "alignSelf": "flex-start",
        "paddingLeft": 30,
        "paddingTop": 10,
      },
      "lightText": Object {
        "color": "#FFFFFF",
        "fontFamily": "ProximaNova-Light",
        "fontSize": 12,
      },
      "semiBoldText": Object {
        "color": "#FFFFFF",
        "fontFamily": "ProximaNova-Semibold",
        "fontSize": 12,
      },
    }
  }
  virtual={undefined}
/>
`;
